waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/mapping/service/impl/PrefixBasedDatabaseMappingService.java

```diff
@@ -93,6 +97,16 @@ private void add(AbstractMetaStore metaStore) {
     mappingsByPrefix.put(metaStoreMapping.getDatabasePrefix(), databaseMapping);
     Whitelist mappedDbWhitelist = new Whitelist(metaStore.getMappedDatabases());
     mappedDbByPrefix.put(metaStoreMapping.getDatabasePrefix(), mappedDbWhitelist);
+
+    List<MappedTables> mappedTables = metaStore.getMappedTables();
+    if (mappedTables != null) {
+      Map<String, Whitelist> mappedTblByDb = new ConcurrentHashMap<>();
```

Comment:
Why does this need to be a `ConcurrentHashMap` instead of just a normal `HashMap`?

Reply:
Hmm, I looked at the other maps in the class and I assumed they all have to be concurrent/synchronised, but since we are not making any changes to `mappedTblByDb` after initialisation, it should be ok to use a normal `HashMap`.

Reply:
I was looking at the `mappedTblByPrefix` above actually, and I think that one doesn't have to be a `ConcurrentHashMap` either?

Reply:
I think the reason that `mappedDbByPrefix` is a `ConcurrentHashMap` applies to `mappedTblByPrefix` as that's a member variable of the class and the `add` method could be called by multiple threads which in turn could update the map concurrently. This doesn't apply to `mappedTblByDb` as it's a method level variable so will be instantiated separately for each caller.

Reply:
Alright, that makes sense